cfg/optimize_test.go

```diff
@@ -0,0 +1,430 @@
+	resetMetadataEndpoints(t)
+	// Create a test server that returns a matching machine type.
+	server := createTestServer(t, func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Metadata-Flavor", "Google")
+		fmt.Fprint(w, "zones/us-central1-a/machineTypes/a3-highgpu-8g")
+	})
+	defer closeTestServer(t, server)
+
+	// Override metadataEndpoints for testing.
+	metadataEndpoints = []string{server.URL}
+	config := defaultOptimizationConfig
+	cfg := &Config{}
+	isSet := &mockIsValueSet{setFlags: map[string]bool{}}
+
+	_, err := applyMachineTypeOptimizations(&config, cfg, isSet)
+
+	assert.NoError(t, err)
+}
+
+func TestSetFlagValue_Bool(t *testing.T) {
+	cfg := &Config{}
+	isSet := &mockIsValueSet{setFlags: map[string]bool{}}
+
+	err := setFlagValue(cfg, "implicit-dirs", flagOverride{newValue: true}, isSet)
+
+	require.NoError(t, err)
+	assert.True(t, cfg.ImplicitDirs)
+}
+
+func TestSetFlagValue_String(t *testing.T) {
+	cfg := &Config{}
+	isSet := &mockIsValueSet{setFlags: map[string]bool{}}
+
+	err := setFlagValue(cfg, "app-name", flagOverride{newValue: "optimal_gcsfuse"}, isSet)
+
+	require.NoError(t, err)
+	assert.Equal(t, "optimal_gcsfuse", cfg.AppName)
+}
+
+func TestSetFlagValue_Int(t *testing.T) {
+	cfg := &Config{}
+	isSet := &mockIsValueSet{setFlags: map[string]bool{}}
+
+	err := setFlagValue(cfg, "metadata-cache.stat-cache-max-size-mb", flagOverride{newValue: 1024}, isSet)
+
+	require.NoError(t, err)
+	assert.EqualValues(t, 1024, cfg.MetadataCache.StatCacheMaxSizeMb)
+}
+
+func TestSetFlagValue_InvalidFlagName(t *testing.T) {
+	cfg := &Config{}
+	isSet := &mockIsValueSet{setFlags: map[string]bool{}}
+
+	err := setFlagValue(cfg, "invalid-flag", flagOverride{newValue: true}, isSet)
+
+	assert.Error(t, err)
+}
+
+func TestApplyMachineTypeOptimizations_NoMachineTypes(t *testing.T) {
+	resetMetadataEndpoints(t)
+	// Create a test server that returns a matching machine type.
+	server := createTestServer(t, func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Metadata-Flavor", "Google")
+		fmt.Fprint(w, "zones/us-central1-a/machineTypes/a3-highgpu-8g")
+	})
+	defer closeTestServer(t, server)
+
+	// Override metadataEndpoints for testing.
+	metadataEndpoints = []string{server.URL}
+
+	config := optimizationConfig{
+		flagOverrideSets: []flagOverrideSet{
+			{
+				name: "high-performance",
+				overrides: map[string]flagOverride{
+					"write.enable-streaming-writes":         {newValue: true},
+					"write.max-concurrency":                 {newValue: 128},
+					"metadata-cache.negative-ttl-secs":      {newValue: 0},
+					"metadata-cache.ttl-secs":               {newValue: -1},
+					"metadata-cache.stat-cache-max-size-mb": {newValue: 1024},
+					"metadata-cache.type-cache-max-size-mb": {newValue: 128},
+					"implicit-dirs":                         {newValue: true},
+					"file-system.rename-dir-limit":          {newValue: 200000},
+					"file-system.gid":                       {newValue: "1000"},
```

Comment:
Remove superfluous code to make test focused - https://testing.googleblog.com/2018/06/testing-on-toilet-keep-tests-focused.html
```suggestion

```